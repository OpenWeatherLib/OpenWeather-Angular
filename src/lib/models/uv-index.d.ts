export interface UvIndex {
  lat: number;
  lon: number;
  date_iso: Date;
  date: Date;
  value: number;
}
