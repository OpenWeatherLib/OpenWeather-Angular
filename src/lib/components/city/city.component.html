<mat-card class="ga-card" *ngIf="city">
  <mat-card-header>
    <img mat-card-avatar class="ga-card-header-image" src="{{cityPictureUrl}}" alt="City pic">
    <mat-card-title>City</mat-card-title>
    <mat-card-subtitle>{{city?.name}}</mat-card-subtitle>
  </mat-card-header>
  <img mat-card-image src="{{cityPictureUrl}}" alt="City pic">
  <mat-card-content>
    <p>Country: {{city?.country}}</p>
    <p>Population: {{city?.population}}</p>
    <p>Coordinates: Lat: {{city?.coord?.lat}}, Lon: {{city?.coord?.lon}}</p>
  </mat-card-content>
</mat-card>
