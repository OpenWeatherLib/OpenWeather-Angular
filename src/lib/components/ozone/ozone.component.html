<mat-progress-bar *ngIf="(isLoading$ | async)" mode="indeterminate"></mat-progress-bar>
<mat-card class="ga-card" *ngIf="(ozone$ | async) && !(isLoading$ | async)">
  <mat-card-header>
    <img mat-card-avatar class="ga-card-header-image" src="/assets/ozone.png" alt="Ozone pic">
    <mat-card-title>Ozone</mat-card-title>
    <mat-card-subtitle>{{(ozone$ | async)?.data}}</mat-card-subtitle>
  </mat-card-header>
  <mat-card-content>
    <p>Coordinates: Lat: {{(ozone$ | async)?.coordinates?.latitude}}, Lon: {{(ozone$ | async)?.coordinates?.longitude}}</p>
  </mat-card-content>
</mat-card>
