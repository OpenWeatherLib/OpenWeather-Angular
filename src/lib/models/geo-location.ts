import { classJsonKey } from "../decorator";

@classJsonKey("", "")
export class GeoLocation {
  lat: number = 720.0;
  long: number = 720.0;

  isDefault(): boolean {
    return this.lat === 720.0 && this.long === 720.0;
  }
}
