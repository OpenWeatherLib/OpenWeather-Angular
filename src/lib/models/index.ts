export * from "./address-component";
export * from "./city";
export * from "./city2";
export * from "./clouds";
export * from "./coordinates";
export * from "./coordinates2";
export * from "./geometry";
export * from "./main";
export * from "./rain";
export * from "./sys";
export * from "./weather-current";
export * from "./weather-forecast";
export * from "./weather-forecast-part";
export * from "./weather-part";
export * from "./wind";
export * from "./uv-index";
