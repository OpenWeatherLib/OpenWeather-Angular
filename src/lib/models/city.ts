import { GeoLocation } from "./geo-location";
import { classJsonKey } from "../decorator";

@classJsonKey("", "")
export class City {
  id: number = 0;
  name: string = "";
  country: string = "";
  population: number = 0;
  geoLocation: GeoLocation = new GeoLocation();

  isDefault(): boolean {
    return this.id === 0 && this.name === "" && this.country === "" && this.population === 0 && this.geoLocation.isDefault();
  }
}
