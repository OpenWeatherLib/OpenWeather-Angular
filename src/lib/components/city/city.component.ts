import { Component, Input, OnInit } from "@angular/core";

import { BaseComponent } from "@lib/components/base-component/base.component";
import { City } from "@lib/models";
import { ImageService, OpenWeatherService } from "@lib/services";

@Component({
  selector: "ga-city",
  templateUrl: "./city.component.html",
  styleUrls: ["./city.component.scss"]
})
export class CityComponent extends BaseComponent implements OnInit {

  @Input() initialCityName: string = String().empty;

  city: City = null;
  cityPictureUrl: string = String().empty;

  constructor(private readonly imageService: ImageService,
    private readonly openWeatherService: OpenWeatherService) {
    super();
  }

  ngOnInit() {
    this.openWeatherService.loadCityData(this.initialCityName);

    this.registerSubscription(
      this.openWeatherService.city()
        .subscribe(city => {
          if (city) {
            this.city = city;
            this.imageService.receiveImagePictureUrl(this.city.name);
          }
        }));

    this.registerSubscription(
      this.imageService.cityPictureUrl()
        .subscribe(url => {
          this.cityPictureUrl = url;
        }));
  }
}
