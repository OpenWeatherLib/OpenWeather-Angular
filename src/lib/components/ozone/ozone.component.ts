import { Component, OnInit } from "@angular/core";

import { BaseComponent } from "@lib/components/base-component/base.component";
import { Ozone } from "@lib/models";
import { OpenWeatherService } from "@lib/services";

@Component({
  selector: "ga-ozone",
  templateUrl: "./ozone.component.html",
  styleUrls: ["./ozone.component.scss"]
})
export class OzoneComponent extends BaseComponent implements OnInit {

  ozone: Ozone = null;

  constructor(private readonly openWeatherService: OpenWeatherService) {
    super();
  }

  ngOnInit() {
    this.registerSubscription(this.openWeatherService.ozone().subscribe(ozone => {
      if (ozone) {
        this.ozone = ozone;
      }
    }));
  }
}
