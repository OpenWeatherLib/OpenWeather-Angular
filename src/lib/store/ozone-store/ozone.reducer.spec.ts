import { City, Ozone } from "@lib/models";
import { loadCitySuccessAction } from "../city-store/city.actions";
import { loadOzoneRequestAction, loadOzoneSuccessAction, loadOzoneErrorAction, setAccuracy, setDateTime } from "./ozone.actions";
import { ozoneReducer } from "./ozone.reducer";
import { OzoneState } from "./ozone.state";

describe("Ozone Reducer Tests", () => {

    const initialState: OzoneState = {
        ozone: undefined,
        dateTime: "2019-07-12",
        accuracy: 2,
        isLoading: false,
        error: "Test Error"
    };

    describe("State Changes", () => {
        test("should have initial state after invalid action", () => {
            // Arrange & Act
            const testState: OzoneState = ozoneReducer(initialState, { type: "INVALID_ACTION" } as any);

            // Assert
            expect(testState).toBe(initialState);
        });

        test("should load city on success of loading", () => {
            // Arrange
            const loadRequest = loadOzoneRequestAction({ city: {} as City });
            const loadSuccess = loadOzoneSuccessAction({ ozone: {} as Ozone });

            // Act
            const testState: OzoneState = [loadRequest, loadSuccess].reduce(ozoneReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });

        test("should set error on error of loading", () => {
            // Arrange
            const loadRequest = loadOzoneRequestAction({ city: {} as City });
            const loadError = loadOzoneErrorAction({ error: "Error" });

            // Act
            const testState: OzoneState = [loadRequest, loadError].reduce(ozoneReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });

        test("should set loading to true on loadCitySuccessAction", () => {
            // Arrange
            const loadSuccess = loadCitySuccessAction({ city: {} as City });

            // Act
            const testState: OzoneState = [loadSuccess].reduce(ozoneReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });

        test("should set accuracy on setAccuracy", () => {
            // Arrange
            const setAccuracyAction = setAccuracy({ accuracy: 7 });

            // Act
            const testState: OzoneState = [setAccuracyAction].reduce(ozoneReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });

        test("should set dateTime on setDateTime", () => {
            // Arrange
            const setDateTimeAction = setDateTime({ dateTime: "DateTime" });

            // Act
            const testState: OzoneState = [setDateTimeAction].reduce(ozoneReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });
    });
});
