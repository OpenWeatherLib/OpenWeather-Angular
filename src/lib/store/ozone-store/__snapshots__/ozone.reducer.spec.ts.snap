// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ozone Reducer Tests State Changes should load city on success of loading 1`] = `
Object {
  "accuracy": 2,
  "dateTime": "2019-07-12",
  "error": undefined,
  "isLoading": false,
  "ozone": Object {},
}
`;

exports[`Ozone Reducer Tests State Changes should set accuracy on setAccuracy 1`] = `
Object {
  "accuracy": 7,
  "dateTime": "2019-07-12",
  "error": "Test Error",
  "isLoading": false,
  "ozone": undefined,
}
`;

exports[`Ozone Reducer Tests State Changes should set dateTime on setDateTime 1`] = `
Object {
  "accuracy": 2,
  "dateTime": "DateTime",
  "error": "Test Error",
  "isLoading": false,
  "ozone": undefined,
}
`;

exports[`Ozone Reducer Tests State Changes should set error on error of loading 1`] = `
Object {
  "accuracy": 2,
  "dateTime": "2019-07-12",
  "error": "Error",
  "isLoading": false,
  "ozone": undefined,
}
`;

exports[`Ozone Reducer Tests State Changes should set loading to true on loadCitySuccessAction 1`] = `
Object {
  "accuracy": 2,
  "dateTime": "2019-07-12",
  "error": "Test Error",
  "isLoading": true,
  "ozone": undefined,
}
`;
